dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/events/trackedentity/AbstractTrackedEntityInstanceService.java

```diff
@@ -720,6 +720,12 @@ private ImportSummary updateTrackedEntityInstance( TrackedEntityInstance dtoEnti
             {
                 String message = "Tracked entity instance " + dtoEntityInstance.getTrackedEntityInstance()
                     + " does not exist";
+
+                if ( teiService.trackedEntityInstanceExistsIncludingDeleted( dtoEntityInstance.getTrackedEntityInstance() ) )
```

Comment:
In this case @gnespolino , you are also saying its not allowed to update a tei. You can update a tei, as long as it is not soft deleted. The null-check you had before did exactly that :) So if you only do the null check, we know that the tei exists in the DB (done earlier in the code) and now it is null meaning it's soft deleted .